gocourse05/security/security.go

```diff
@@ -0,0 +1,104 @@
+package security
+
+type Rotater interface {
+	Rotate(angle float32)
+}
+
+type Recorder interface {
+	Record(c Conditions, records *[]AnimalRecord, rec AnimalRecord)
+}
+
+type AnimalRecord struct {
+	AnimalType
+	AnimalAction
+}
+
+func NewAnimalRecord(t AnimalType, a AnimalAction) *AnimalRecord {
+	return &AnimalRecord{
+		AnimalType:   t,
+		AnimalAction: a,
+	}
+}
+
+type AnimalType string
+type AnimalAction string
+
+const (
+	Bear  AnimalType = "Bear"
+	Tiger AnimalType = "Tiger"
+	Rat   AnimalType = "Rat"
+
+	Sleep   AnimalAction = "Sleep"
+	Eat     AnimalAction = "Eat"
+	Walking AnimalAction = "Walking"
+)
+
+type Conditions struct {
+	IsNight      bool
+	DistToAnimal float32
+	Battery      int
+}
+
+func NewConditions(isNight bool, dst float32, battery int) *Conditions {
+	return &Conditions{
+		IsNight:      isNight,
+		DistToAnimal: dst,
+		Battery:      battery,
+	}
+}
+
+type PTZCamera struct {
+	Range float32
+	Angle float32
```

Comment:
Давай зробимо інкапсуляцію цих полей:
```suggestion
	range float32
	angle float32
```

Reply:
Для `FixedCamera` також.